### ISSUE TITLE
Confused about the input width and height

### ISSUE THREAD

**Issue by xichen714** [jwagner/smartcrop.js#80]
Hi,

Since the result topCrop is the relative crop area, is there any difference between setting w = 100, h = 100 and w = 200, h = 200? Because sometimes I don't know the size of the images, and if I set the width and height too large, smartCrop cannot work. If I set the width and height too small, things may be chopped off. Since I have many images, if I set width and height for each image according to their size, it would be too slow.

Thank you so much!

Best,
Xi

**Comment by jwagner** [comment]
> Since the result topCrop is the relative crop area, is there any difference between setting w = 100, h = 100 and w = 200, h = 200?

There are a few differences, for instance smartcrop will never suggest a crop that would require upscaling when using a custom minScale. If the dimensions provided are smaller than 256x256 the prescaling before analyzing the image will be to the actual output resolution rather than the maximum of 256 for each side.

**TL;DR** it's details, the most important bit is that you get the aspect ratio right and then crop the image according to the data provided by smartcrop.
